js/node-manager/src/client/screens/Settings.tsx

```diff
@@ -1,14 +1,31 @@
 import { useAppState } from '../app-state'
 import { Button } from '../components/basics'
-import { Input, Layout } from '../components/Layout'
+import { Layout } from '../components/Layout'
 import React, { useEffect, useMemo, useState } from 'react'
 import AceEditor from 'react-ace'
 import clsx from 'clsx'
 import { NodeType, ReachableNodeUi } from '../../common/types'
 import { get, set, parse } from 'json-pointer'
-import deepEqual from 'deep-equal'
 import { validateAgainstSchema } from '../components/SettingsEditor'
 
+// Handles case where `set` cannot set root path
+// Because require('json-pointer').set throws when `path` is empty
+const setOrReplace = (s: object, path: string, val: unknown): object => {
+  // parse:
+  //  "/a/b" => ['a', 'b']
+  //  "/" => []
+  //  "" => []
+  const pointer = parse(path.trim()).filter((x) => !!x)
+  if (pointer.length === 0) {
+    const c: { default: object } = { default: s }
+    set(c, '/default', val)
+    return c.default
```

Comment:
> Could you explain what's going on "behind the scenes" here?

I don't want to get into the implementation details of https://www.npmjs.com/package/json-pointer, so I can't.

There may be subvalue merging inside. It might be just purely a replacement. 
But by using `set` instead of returning, I can be sure that it will behave the same as the `else` branch.